parser_library/src/workspaces/workspace.cpp

```diff
@@ -134,28 +134,46 @@ void workspace::collect_diags() const
         collect_diags_from_child(*pfc.m_processor_file);
 }
 
+namespace {
+void generate_merged_fade_messages(
+    const std::string& uri, const processing::hit_count_details& hc_details, std::vector<fade_message_s>& fms)
+{
+    const auto line_hits_it_b = hc_details.line_hits.begin();
+    const auto it_e = std::next(line_hits_it_b, hc_details.max_lineno + 1);
+    const auto it_b =
+        std::find_if(line_hits_it_b, it_e, [](const processing::hit_count& e) { return e.contains_statement; });
+
+    static constexpr auto hit_or_no_stmt_predicate = [](const processing::hit_count& e) {
+        return e.count != 0 || !e.contains_statement;
+    };
+
+    auto it = std::find_if_not(it_b, it_e, hit_or_no_stmt_predicate);
+    while (it != it_e)
+    {
+        auto last_inactive_line = std::find_if(it, it_e, hit_or_no_stmt_predicate);
+        if (last_inactive_line != it_e || it != it_b || !hc_details.is_external_macro)
+            fms.emplace_back(fade_message_s::inactive_statement(uri,
+                range(position(std::distance(line_hits_it_b, it), 0),
+                    position(std::distance(line_hits_it_b, std::prev(last_inactive_line)), 72))));
+
+        it = std::find_if_not(last_inactive_line, it_e, hit_or_no_stmt_predicate);
+    }
+}
```

Comment:
Could you reformulate this code with a predicate `faded_statement` (i.e. `e.contains_statement && e.count == 0`)?

Reply:
or `faded_line`

Reply:
btw I suggest switching from 72 to either 80 or something like -1...
72 creates a strange visual artefact where the last line in a block is partially faded when you have the sequence numbers in.

Reply:
Reformulated. I ultimately opted for `80` as it was weird seeing the big number being present in `fade_message_s` range